pallets/runtime/tests/src/settlement_test.rs

```diff
@@ -3066,3 +3083,371 @@ fn multiple_custodian_settlement() {
+    // These are the encryptions keys used by MERCAT and are different from the signing keys
+    // that Polymesh uses.
+    let mut rng = StdRng::from_seed(seed);
+    let elg_secret = ElgamalSecretKey::new(Scalar::random(&mut rng));
+    let elg_pub = elg_secret.get_public_key();
+    let enc_keys = EncryptionKeys {
+        pblc: elg_pub.into(),
+        scrt: elg_secret.into(),
+    };
+
+    let asset_id = AssetId {
+        id: *Ticker::try_from(token_name).unwrap().as_bytes(),
+    };
+    let asset_id_witness = CommitmentWitness::from((asset_id.clone().into(), &mut rng));
+    let secret_account = SecAccount {
+        enc_keys,
+        asset_id_witness,
+    };
+    let valid_asset_ids = convert_asset_ids(valid_asset_ids);
+    let mercat_account_tx = AccountCreator {}
+        .create(tx_id, &secret_account, &valid_asset_ids, &mut rng)
+        .unwrap();
+
+    (secret_account, mercat_account_tx)
+}
+
+pub fn init_account(
+    tx_id: u32,
+    seed: [u8; 32],
+    token_name: &[u8],
+    owner: Public,
+) -> (SecAccount, EncryptedAssetIdWrapper) {
+    let valid_asset_ids: Vec<AssetId> = ConfidentialAsset::confidential_tickers();
+    let (scrt_account, mercat_account_tx) = gen_account(tx_id, seed, token_name, valid_asset_ids);
+
+    ConfidentialAsset::validate_mercat_account(Origin::signed(owner), mercat_account_tx.clone())
+        .unwrap();
+
+    (
+        scrt_account,
+        EncryptedAssetIdWrapper::from(mercat_account_tx.pub_account.enc_asset_id.encode()),
+    )
+}
+
+pub fn create_account_and_mint_token(
+    owner: Public,
+    owner_did: IdentityId,
+    total_supply: u128,
+    token_name: Vec<u8>,
+    tx_id: u32,
+    seed: [u8; 32],
+) -> u32 {
+    let funding_round_name: FundingRoundName = b"round1".into();
+
+    let token = SecurityToken {
+        name: token_name.clone().into(),
+        owner_did,
+        total_supply: total_supply,
+        divisible: false,
+        asset_type: AssetType::default(),
+        primary_issuance_agent: Some(owner_did),
+        ..Default::default()
+    };
+    let ticker = Ticker::try_from(token.name.as_slice()).unwrap();
+    let identifiers = vec![(IdentifierType::default(), b"undefined".into())];
+
+    assert_ok!(ConfidentialAsset::create_confidential_asset(
+        Origin::signed(owner),
+        token.name.clone(),
+        ticker,
+        true,
+        token.asset_type.clone(),
+        identifiers.clone(),
+        Some(funding_round_name.clone()),
+    ));
+
+    // In the initial call, the total_supply must be zero.
+    assert_eq!(Asset::token_details(ticker).total_supply, Zero::zero());
+
+    // ---------------- prepare for minting the asset
+
+    let valid_asset_ids: Vec<AssetId> = ConfidentialAsset::confidential_tickers();
+
+    let (scrt_account, mercat_account_tx) = gen_account(tx_id, seed, &token_name, valid_asset_ids);
+
+    ConfidentialAsset::validate_mercat_account(Origin::signed(owner), mercat_account_tx.clone())
```

Comment:
Minor remark but the lint checker would have suggested `let _ =` here.
Also it looks strange that the result of validation is not used. You could use `assert_ok` for example to ensure that the result is not an `Err` instead of unwrapping.